### PR TITLE
[query/pcode] Remove IRInterval from IntervalFunctions

### PR DESCRIPTION
**Pull request by chrisvittal**
It was basically a proto PIntervalValue

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala [hunk 1]
```diff
@@ -82,54 +82,71 @@ object IntervalFunctions extends RegistryFunctions {
...
-        Code(
-          interval1.storeToLocal,
-          interval2.storeToLocal,
-          !(interval1.isEmpty || interval2.isEmpty ||
-            interval1.isBelowOnNonempty(interval2) ||
-            interval1.isAboveOnNonempty(interval2))
-        )
+          interval1.isEmpty(cb) || interval2.isEmpty(cb) ||
+            isBelowOnNonempty(cb, interval1, interval2) ||
+            isAboveOnNonempty(cb, interval1, interval2)
```
Comment (johnc1231) [inline hail/src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala:147]: I think this is going to result in generating the setup code for the start and end of both intervals twice. I'm not sure if that's a bad thing, but wanted to flag it and ask.

**Review by johnc1231 — CHANGES_REQUESTED** [review]

File: hail/src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala:147]: it's memoized on line 121, this should be fine I think

Comment (johnc1231) [reply hail/src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala:147]: So I see that it's memoized, but then it's converted into an `EmitCode` and the `setup` is added to the code builder in both methods. Is the expectation that the `setup`s are empty (or at least tiny) for something that is coming from a memoized value?

Comment (chrisvittal) [reply hail/src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala:147]: 1. EmitCode.fromI currently always produces an empty setup.
2. This should be no worse than the old code in that it also does a double load, once for the `isEmpty`, and again for the `isAbove/isBelow` calls. An extra memoize to turn these loads into variable loads could be done, but I feel it's largely unnecessary.

Comment (johnc1231) [reply hail/src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala:147]: Just wanted to make sure I understood. Reasoning makes sense

**Comment by johnc1231** [comment]
Compilation errors

**Pull request merged**